### ISSUE TITLE
low priority bug: parallel std out from each simulation stumple each other

### ISSUE THREAD

**Issue by fdobad** [fire2a/C2F-W#63]
when running multiple simulations in parallel, the standard output (stdout) from each simulation is getting mixed up or interleaved, making it difficult to read and understand the output from each individual simulation.

- [ ] imprimir en paquetes o buffers por simulacion para no perder la logica de la tabla resumen, ejemplo:
```

Selected (Random) ignition point for Year 1, sim 2: 160673
----------------------------- Results -----------------------------
Total Available Cells: 37060 - % of the Forest: 9.4342%
Total Burnt Cells: 113977 - % of the Forest: 29.0146%
Selected (Random) ignition point for Year 1, sim 1: 273808 < --- !!!!
----------------------------- Results ----------------------------- < --- !!!!
Total Available Cells: 25980 - % of the Forest: 6.61362% < --- !!!!
Total Non-Burnable Cells: 241789 - % of the Forest: 61.5512%
Total Firebreak Cells: 0 - % of the Forest: 0%


Total Burnt Cells: 125057 - % of the Forest: 31.8352%
Total Non-Burnable Cells: 241789 - % of the Forest: 61.5512%
Total Firebreak Cells: 0 - % of the Forest: 0%
Cell2Fire simulator finished.
```
- [ ] indicar cuánto falta % simulaciones hechas / total de simulaciones

**Comment by fdobad** [comment]
See IgnitionsHistory/Ignitions.csv for replacement.

**Comment by matildeRivas** [comment]
Create a printer class and use mutex for cout.
This will probably slow down the simulations a bit since threads will have to wait for the mutex to be released so we should add an option for "debugging logs" (verbose level 1? @fdobad )
